Docs/Pre v1 propose refactor

jlnav opened the pull request:
Addresses #1012 

There's a couple ideas I'd like to address, primarily for **beginning users**:
- Make small, simple code samples viewable as soon as possible
- Flexibility is important, but shouldn't supersede ease-of-use or ease-of-initial-understanding

> - Is it possible to avoid this common docs pitfall: "if doing this, then this, if doing that, do that, if doing this with some of that do part of this and that."
> - Show one simple way of doing something first, then alternatives come later

- Move, or remove, any "justification for existence"
- Maybe pick a new theme? It's a one-line change: https://sphinx-themes.org/

shuds13 commented on docs/tutorials/executor_forces_tutorial.rst:
```diff
@@ -371,15 +372,79 @@ These may require additional browsing of the documentation to complete.
             task.poll()
 
-            if task.file_exists_in_workdir(statfile):
+            if task.file_exists_ins_workdir(statfile):
```
?

jlnav replied:
Whoops, must have been a weird find-and-replace

jmlarson1 reviewed (approved):
Thank you so much for the docs edits @jlnav !

jlnav commented:
I recognize and apologize that this docs branch is large and covers too many files. As requested, I've selected highlights to simplify reviewing:

1. Fixed and updated links and refs throughout
2. LibeSpecs parameters/settings adjusted/corrected
3. Many sections involving multiple options use tabs
installation BEFORE: https://libensemble.readthedocs.io/en/develop/advanced_installation.html
installation AFTER: https://libensemble--992.org.readthedocs.build/en/992/advanced_installation.html
comms BEFORE: https://libensemble.readthedocs.io/en/develop/running_libE.html
comms AFTER: https://libensemble--992.org.readthedocs.build/en/992/running_libE.html
4. Many pages with lots/long subsections use dropdowns
FAQ BEFORE: https://libensemble.readthedocs.io/en/develop/FAQ.html
FAQ AFTER: https://libensemble--992.org.readthedocs.build/en/992/FAQ.html
batch scripts BEFORE: https://libensemble.readthedocs.io/en/develop/platforms/example_scripts.html
batch scripts AFTER: https://libensemble--992.org.readthedocs.build/en/992/platforms/example_scripts.html
4. Many introductory sections made more succinct
understanding BEFORE: https://libensemble.readthedocs.io/en/develop/overview_usecases.html
understanding AFTER: https://libensemble--992.org.readthedocs.build/en/992/overview_usecases.html
6. Stefan's Suggestions
docstrings end with periods, "Optional" and "Boolean" always capitalized. Subsection titles capitalized. "where" lowercased when used to followup a code-sample.
7. Function Guides slimmed down
index BEFORE: https://libensemble.readthedocs.io/en/develop/function_guides/function_guide_index.html
index AFTER: https://libensemble--992.org.readthedocs.build/en/992/function_guides/function_guide_index.html
gens BEFORE: https://libensemble.readthedocs.io/en/develop/function_guides/generator.html
gens AFTER: https://libensemble--992.org.readthedocs.build/en/992/function_guides/generator.html
- Show signature/code samples first
- When listing multiple components, or describing orders-of-operations, prioritize numbered-lists over paragraphs

8. First tutorial translated to OO, each step made a tab, intro-section slimmed down
BEFORE: https://libensemble.readthedocs.io/en/develop/tutorials/local_sine_tutorial.html
AFTER: https://libensemble--992.org.readthedocs.build/en/992/tutorials/local_sine_tutorial.html

shuds13 commented:
@jlnav Can I request to stop changing the theme, makes it harder to focus on content. I think we should stick with same theme for now anyway.

shuds13 commented:
Need to add sphinx-copybutton to setup.py (under "docs")

shuds13 commented on docs/platforms/platforms_index.rst:
```diff
@@ -213,7 +213,7 @@ libEnsemble on specific HPC systems.
     perlmutter
     polaris
-    spock/crusher <spock_crusher>
+    Spock/Crusher <spock_crusher>
```
Why is this capitalized (while not the others)?

jlnav replied:
Otherwise the system names appear lowercased: https://libensemble.readthedocs.io/en/main/platforms/platforms_index.html#instructions-for-specific-platforms

But since `spock_crusher.rst` has the properly titled names, listing that file in the toctree without overwriting its title should be fine. Just tested locally. I'll make the simple change

shuds13 commented on docs/programming_libE.rst:
```diff
@@ -4,23 +4,29 @@ Constructing Workflows
+   :maxdepth: 2
+   :caption: Helpful Additions
+
+   function_guides/history_array
```
I still think of history array as a central feature of libE, not a helpful addition.

jmlarson1 replied:
I agree

jlnav replied:
👍
